src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationIconValueProvider.cs

```diff
@@ -73,8 +73,32 @@ internal sealed class ApplicationIconValueProvider : InterceptingPropertyValuePr
                 propertyValue = PathHelper.MakeRelative(_unconfiguredProject, unevaluatedPropertyValue);
             }
 
-            await _sourceItemsProvider.AddAsync(Content.SchemaName, propertyValue);
+            string existingPropertyValue = await defaultProperties.GetEvaluatedPropertyValueAsync(ConfigurationGeneral.ApplicationIconProperty);
+            IProjectItem? existingItem = await GetExistingContentItemAsync(existingPropertyValue);
```

Comment:
I don't think ISourceItemsProvider.AddAsync will add a duplicate item for an item that already exists - have you confirmed that it does and you had to work around that?

Reply:
Ah, I see you are handling the replacing part.